src/dataconnect/schemaMigration.ts

```diff
@@ -189,6 +194,66 @@ export async function migrateSchema(args: {
   return [];
 }
 
+export async function grantRoleToUserInSchema(options: Options, schema: Schema) {
+  const { serviceName, instanceId, instanceName, databaseId } = getIdentifiers(schema);
+  await ensureServiceIsConnectedToCloudSql(
+    serviceName,
+    instanceName,
+    databaseId,
+    /* linkIfNotConnected=*/ false,
+  );
+
+  const role = options.role as string;
+  const email = options.email as string;
+  if (!role) {
+    throw new FirebaseError(
+      "-R, --role <role> is required. Run the command with -h for more info.",
+    );
+  }
+  if (!email) {
+    throw new FirebaseError(
+      "-E, --email <email> is required. Run the command with -h for more info.",
+    );
+  }
+
+  // Make sure current user can perform this action.
+  const userIsCSQLAdmin = await iamUserIsCSQLAdmin(options);
+  if (!userIsCSQLAdmin) {
+    throw new FirebaseError(`Only users with 'roles/cloudsql.admin' can grant SQL roles.`);
```

Comment:
This might be a good place to print out instructions telling users to ask their DB admin - ie:
```
Only users with 'roles/cloudsql.admin' can grant SQL roles. If you do not have this role, ask your database administrator to run this command or manually grant <role> to <user>
```

Reply:
Updated the error message. The error message also will take care of the service account extension.